### PR TITLE
[cuda] Drop cuda1 HAL implementation code

### PR DESCRIPTION
**Pull request by antiagainst**
Now cuda2 is the default; we can drop cuda1 to reduce maintenance.

Related to https://github.com/openxla/iree/issues/13245

### CODE REVIEW

**Review by benvanik — APPROVED** [review]

**Comment by benvanik** [comment]
champagne-popping-cork-blow-4zoaopphhcna4bp6 1

**Comment by antiagainst** [comment]
I'll stage it here for some time and land likely end of the week. :)

File: tests/e2e/stablehlo_ops/CMakeLists.txt [hunk 1]
```diff
@@ -302,170 +302,6 @@ iree_check_single_backend_test_suite(
     "local"
 )
 
-iree_check_single_backend_test_suite(
-  NAME
-    check_cuda_graph
```
Comment (ScottTodd) [inline tests/e2e/stablehlo_ops/CMakeLists.txt:307]: FYI, my recent https://github.com/openxla/iree/pull/16086 is working as intended here :D

When 'cuda' was enabled and 'cuda2' was disabled: https://github.com/openxla/iree/actions/runs/7632069510/job/20791981080#step:7:1203
```
The following tests did not run:
	593 - iree/tests/e2e/stablehlo_ops/check_cuda2_graph_abs.mlir (Disabled)
	594 - iree/tests/e2e/stablehlo_ops/check_cuda2_graph_add.mlir (Disabled)
```

After the flip, when 'cuda2' was enabled' and 'cuda' was disabled:
https://github.com/openxla/iree/actions/runs/7632476397/job/20793221872#step:7:1221
```
The following tests did not run:
	471 - iree/tests/e2e/stablehlo_ops/check_cuda_graph_abs.mlir (Disabled)
	472 - iree/tests/e2e/stablehlo_ops/check_cuda_graph_add.mlir (Disabled)
```

Prior to that change, the disabled targets wouldn't appear in ctest output at all. Now it is pretty clear what is enabled or disabled. (I could have checked that on the flip PR itself, but I forgot :P)

**Review by ScottTodd — APPROVED** [review]

File: tests/e2e/stablehlo_ops/CMakeLists.txt (hunk 1 above)
Comment (antiagainst) [reply tests/e2e/stablehlo_ops/CMakeLists.txt:307]: Ah, nice to know!

**Pull request merged**